platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheIntegrationTest.groovy

```diff
@@ -36,6 +36,41 @@ class ConfigurationCacheIntegrationTest extends AbstractConfigurationCacheIntegr
         !output.contains("Configuration cache is an incubating feature.")
     }
 
+    def "should not create an entry on a cache miss if in read-only mode"() {
+        def configurationCache = newConfigurationCacheFixture()
+
+        given:
+        settingsFile << ""
+
+        when:
+        run("help", "--configuration-cache", "-Dorg.gradle.configuration-cache.internal.read-only=true")
```

Comment:
💅 `configurationCacheRun("help", "-Dorg.gradle.configuration-cache.internal.read-only=true")`?
Same for other executor invocations